src/pages/PrivateNotes/PrivateNotesViewPage.js

```diff
@@ -66,6 +66,7 @@ function PrivateNotesViewPage({route, personalDetailsList, session, report}) {
             >
                 <HeaderWithBackButton
                     title={translate('privateNotes.title')}
+                    onBackButtonPress={() => Navigation.goBack(ROUTES.PRIVATE_NOTES_LIST.getRoute(report.reportID))}
```

Comment:
This is tricky because we will conditionally show the list page. I was working on this here https://github.com/Expensify/App/pull/28105 but couldn't prioritize it. 

In case a user has a single note that they can access we will take them right away to the view page. So you can't always be sure that view page would go back to list page.

Navigaton flow:
Single note: `Private notes button` -> view page -> edit page
Multiple notes: `Private notes button` -> list page -> view page -> edit page

Reply:
Navigating them back gets even trickier because you need to know which page they came from(it could be the profile page or the report details page). That's why I sort of just relied on the `goBack` default behaviour to sort that out from the navigation stack.

Reply:
Right, so we shall only change the `PrivateNotesEditPage` as we are sure that user would have came from `PrivateNotesViewPage`, right?

Reply:
No no, have a look at the navigation flow I have mentioned here https://github.com/Expensify/App/pull/28273#discussion_r1340757961

Navigaton flow:
Single note: Private notes button -> view page -> edit page
Multiple notes: Private notes button -> list page -> view page -> edit page

So it is between the private notes button and the view page.

Reply:
PrivateNotesEditPage would always be accessed from the view page. It is the view page that is tricky. You can access them right away if you just have a single accessible note or through list page if you have multiple accessible notes.

Reply:
I think you misunderstood me here https://github.com/Expensify/App/pull/28273#discussion_r1340827792

I was talking about whole PR, so we are only sure that if we are on PrivateNotesEditPage then we have came from PrivateNotesView page.

So as part of this PR we shall only change `PrivateNotesEditPage` and revert changes for `PrivateNotesViewPage`.

Please tell if I am getting this right

Reply:
> I think you misunderstood me here https://github.com/Expensify/App/pull/28273#discussion_r1340827792

Aaah, yes sorry, I went with negation. Since you are fixing the navigation behaviour here I would say we also want to fix for the View page, can you look into it to see if we have a way?

Reply:
I am also sort of confused, why we don't retain the navigation stack when user refreshes the page. Does that happen with all the pages?

Reply:
Yes, I have seen same behaviour for workspace pages that we have added fallback route.

Reply:
@techievivek Can you tell me scenarios where I can see multiple private notes. Because currently I can only see one in workspace as well.

Reply:
@techievivek I would like to use logic that you have used https://github.com/Expensify/App/pull/28105/files#diff-65c096044d5f69b35bcdec14c99c4fda4580759df9b1a7c36650d58eea276f1dR3569 here.

To check number of reports and based on that navigate. Let me know if this sounds right to you